### PR TITLE
Full support for 3D biomes

### PR DESCRIPTION
**Pull request by dordsor21**
Since we're only supporting 1.15+ there's no need to try anything other than compatibility

This will currently break any existing biome history (and biome history only) files.

### CODE REVIEW

**Comment by SpeedRider** [comment]
oooo can't wait for such a cool thing.

**Review by N0tMyFaultOG — APPROVED** [review]

File: worldedit-core/src/main/java/com/boydti/fawe/object/extent/OffsetExtent.java [hunk 1]
```diff
@@ -46,10 +43,4 @@ public boolean setBiome(int x, int y, int z, BiomeType biome) {
         throws WorldEditException {
         return getExtent().setBlock(x + dx, y + dy, z + dz, block);
     }
-
-    @Override
-    public ResettableExtent setExtent(Extent extent) {
-        mutable = new MutableBlockVector2();
-        return super.setExtent(extent);
-    }
```
Comment (MattBDev) [inline worldedit-core/src/main/java/com/boydti/fawe/object/extent/OffsetExtent.java:54]: What was the reason for removing this?

File: worldedit-core/src/main/java/com/boydti/fawe/object/clipboard/DiskOptimizedClipboard.java [hunk 2]
```diff
@@ -185,7 +185,7 @@ public boolean setBiome(BlockVector3 position, BiomeType biome) {
 
     @Override
     public boolean setBiome(int x, int y, int z, BiomeType biome) {
-        setBiome(getIndex(x, y, z), biome);
+        setBiome(getIndex(x, 0, z), biome);
```
Comment (MattBDev) [inline worldedit-core/src/main/java/com/boydti/fawe/object/clipboard/DiskOptimizedClipboard.java:188]: Why was this changed from `y` to `0`?

File: worldedit-core/src/main/java/com/sk89q/worldedit/extent/OutputExtent.java [hunk 3]
```diff
@@ -94,7 +94,11 @@ default boolean fullySupports3DBiomes() {
      */
     @Deprecated
     default boolean setBiome(BlockVector2 position, BiomeType biome) {
-        return setBiome(position.toBlockVector3(), biome);
+        boolean result = false;
+        for (int y = 0; y < 256; y ++) {
+            result |= setBiome(position.toBlockVector3().mutY(y), biome);
```
Comment (MattBDev) [inline worldedit-core/src/main/java/com/sk89q/worldedit/extent/OutputExtent.java:99]: Not ideal but I can't think of a better solution right now and it works.

**Comment by MattBDev** [comment]
Looks like I was super lazy when I tried doing 3D Biomes the first time. Oops.

File: worldedit-core/src/main/java/com/boydti/fawe/object/extent/OffsetExtent.java (hunk 1 above)
Comment (dordsor21) [reply worldedit-core/src/main/java/com/boydti/fawe/object/extent/OffsetExtent.java:54]: 3D biomes means no need for the mutable vector so it doesn't need to be reset. Technically the extent doesn't even need to be resettable anymore so I'll change that as well.

File: worldedit-core/src/main/java/com/boydti/fawe/object/clipboard/DiskOptimizedClipboard.java (hunk 2 above)
Comment (dordsor21) [reply worldedit-core/src/main/java/com/boydti/fawe/object/clipboard/DiskOptimizedClipboard.java:188]: because it's broken otherwise. Clipboards do not have 3D biome support yet, and it was a mistake for the y to be there in the first place (as stated in the commit for the change).

File: worldedit-core/src/main/java/com/sk89q/worldedit/extent/OutputExtent.java (hunk 3 above)
Comment (dordsor21) [reply worldedit-core/src/main/java/com/sk89q/worldedit/extent/OutputExtent.java:99]: This is really the only solution; biomes are only stored in 4x4x4 sections, however, doing every 4 blocks could and would cause issues with offsets/transforms, etc

**Review by MattBDev — APPROVED** [review]
